components/director/internal/domain/eventapi/repository.go

```diff
@@ -12,17 +15,89 @@ func NewRepository() *inMemoryRepository {
 	return &inMemoryRepository{store: make(map[string]*model.EventAPIDefinition)}
 }
 
-func (inMemoryRepository) ListByApplicationID(applicationID string) ([]*model.EventAPIDefinition, error) {
-	//TODO panic("implement me")
-	return nil, nil
+func (r *inMemoryRepository) GetByID(id string) (*model.EventAPIDefinition, error) {
+	if api, ok := r.store[id]; ok {
+		return api, nil
+	}
+
+	return nil, errors.Errorf("APIDefinition with %s ID does not exist", id)
```

Comment:
I think returns in this function should be swapped so that we return error early and last return is a success.

Reply:
done